### PR TITLE
[FIX]output_modules.php(message_headers): Show Snooze and Block Sender options....

### PR DESCRIPTION
**Pull request by Baraka24**
- [FIX]output_modules.php(message_headers): Show Snooze and Block Sender options for non draft m

essage only
 - @marclaporte wrote: 

> I am in a viewing a draft email. Why would I want to snooze or block sender?
cyphttask

### CODE REVIEW

File: modules/imap/output_modules.php [hunk 1]
```diff
@@ -370,27 +370,31 @@ protected function output() {
             $txt .= ' | <a class="hlink" id="copy_message" href="#">'.$this->trans('Copy').'</a>';
             $txt .= ' | <a class="hlink" id="move_message" href="#">'.$this->trans('Move').'</a>';
             $txt .= ' | <a class="archive_link hlink" id="archive_message" href="#">'.$this->trans('Archive').'</a>';
-            $txt .= ' | ' . snooze_dropdown($this, isset($headers['X-Snoozed']));
+            if (!isset($headers['Flags']) || !stristr($headers['Flags'], 'draft')) {
```
Comment (josaphatim) [inline modules/imap/output_modules.php:373]: Why not create a variable named for example is draft which will hold this value

`$is_draft = isset($headers['Flags']) && stristr($headers['Flags'], 'draft');`

Then test with is draft in multiple if that are below

File: modules/imap/output_modules.php [hunk 2]
```diff
@@ -370,27 +370,31 @@ protected function output() {
...
-                    $sender = addr_parse($headers['From'])['email'];
-                    $domain = '*@'.get_domain($sender);
-                    $blocked_senders = get_blocked_senders_array($imap_server, $this->get('site_config'), $this->get('user_config'));
-                    $sender_blocked = in_array($sender, $blocked_senders);
-                    $domain_blocked = in_array($domain, $blocked_senders);
-                    if(!in_array($sender, $existing_emails)){
-                        $txt .= ' | <div class="dropdown d-inline-block"><a class="block_sender_link hlink dropdown-toggle'.($domain_blocked || $sender_blocked ? '" id="unblock_sender" data-target="'.($domain_blocked? 'domain':'sender').'"' : '"').' href="#" aria-labelledby="dropdownMenuBlockSender" data-bs-toggle="dropdown"><i class="bi bi-lock-fill"></i> <span id="filter_block_txt">'.$this->trans($domain_blocked ? 'Unblock Domain' : ($sender_blocked ? 'Unblock Sender' : 'Block Sender')).'</span></a>';
-                        $txt .= block_filter_dropdown($this);
+            if (!isset($headers['Flags']) || !stristr($headers['Flags'], 'draft')) {
```
Comment (josaphatim) [inline modules/imap/output_modules.php:378]: this is uncessary you can add that condition to the if below ($this->get('sieve_filters_enabled'))
`if ($this->get('sieve_filters_enabled') && !$is_draft)`

**Comment by Baraka24** [comment]
Thanks, @josaphatim for the review... I have applied all changes as required.

File: modules/imap/output_modules.php [hunk 3]
```diff
@@ -370,29 +370,30 @@ protected function output() {
...
+                    } else {
+                        $txt .= ' | <span data-bs-toogle="tooltip" title="This functionality requires the email server support &quot;Sieve&quot; technology which is not provided. Contact your email provider to fix it or enable it if supported."><i class="bi bi-lock-fill"></i> <span id="filter_block_txt">'.$this->trans('Block Sender').'</span></span>';
                     }
-                } else {
-                    $txt .= ' | <span data-bs-toogle="tooltip" title="This functionality requires the email server support &quot;Sieve&quot; technology which is not provided. Contact your email provider to fix it or enable it if supported."><i class="bi bi-lock-fill"></i> <span id="filter_block_txt">'.$this->trans('Block Sender').'</span></span>';
-                }
             }
+            $txt .= ' | <a class="hlink" id="show_message_source" href="#">' . $this->trans('Show Source') . '</a>';
 
             if (isset($headers['Flags']) && stristr($headers['Flags'], 'draft')) {
```
Comment (josaphatim) [inline modules/imap/output_modules.php:398]: You must use $is_draft variable also here

**Pull request merged**